### PR TITLE
Fix truncated SCRIPT_NAME

### PR DESCRIPTION
**Pull request by MGatner**
**Description**
It appears that some PHP web servers do not guarantee the `SCRIPT_NAME` as a substring of `REQUEST_URI`. From consulting other frameworks, it is still safe to determine the relative path this way, but we need to ensure the segments before checking.

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPdocs
- [X] Unit testing, with >80% coverage
- n/a User guide updated
- [X] Conforms to style guide

### CODE REVIEW

File: system/Language/en/Publisher.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+// Publisher language settings
+return [
+	'expectedFile'      => 'Publisher::{0} expects a valid file.',
+	'expectedDirectory' => 'Publisher::{0} expects a valid directory.',
+	'collision'         => 'Publisher encountered an unexpected {0} while copying {1} to {2}.',
+];
```
Comment (samsonasik) [inline system/Language/en/Publisher.php:17]: @MGatner it seems the new files are unrelated to the PR?

Comment (MGatner) [reply system/Language/en/Publisher.php:17]: Wow yeah sorry about that! I think I forgot to `-u` my stash so the new files carried over. I will force back to the correct file.

**Review by samsonasik — APPROVED** [review]

**Review by paulbalandan — APPROVED** [review]
